docs/concepts/architecture.md

```diff
@@ -10,18 +10,21 @@ Raccoon was built with a primary purpose to source or collect user behaviour dat
 
 At a high level, the following sequence details the architecture.
 
-* Clients make websocket connections to Raccoon by performing a http GET API call, with headers to upgrade to websocket.
-* Raccoon uses [gorilla websocket](https://github.com/gorilla/websocket) handlers and for each connection the handlers spawn a goroutine to handle incoming requests.
-* After the connection has been established, clients can send the events. 
-* Clients can send the request anytime as long as the websocket connection is alive.
-* When the events are available to be consumed, the handler will deserialize the proto events and forward them to the buffered channel.
+* Clients make wiebsocket connections to Raccoon by performing a http GET API call, with headers to upgrade to websocket.
+* Clients connecting via REST endpoint connect to Racccon by requesting the same websocket HTTP server but with POST method.
+* Clients using gRPC need to generate the grpc client for the [EventService](https://github.com/odpf/proton/blob/main/odpf/raccoon/EventService.proto)
+* Raccoon uses [gorilla websocket](https://github.com/gorilla/websocket) handlers and for each websocket connection the handlers spawn a goroutine to handle incoming requests.
+* After the websocket connection has been established, clients can send the events. 
+* Clients can send the request anytime as long as the websocket connection is alive whereas with REST and gRPC requests can be sent only once.
+* When the events are available to be consumed, the handler will deserialize events using the correct deserializer and forward them to the buffered channel.
```

Comment:
We need to restructure this to make it independent of the protocol. We can do something like
* Raccoon accept request from one of the protocols [accepted protocols]
* The event is forwarded to buffer channel
* Acknowledgment is returned to the client
* Each worker iterates over the events' batch buffer, determines the topic based on the type and serializes the bytes to the Kafka producer synchronously.

Or, you can also put detail of each of the protocol connections on this doc like this.
* Raccoon accept request from one of following protocol
   * WS, initiate connection -> maintain connection -> push event
   * GRPC ......
   * .......
* The event is forwarded to buffer channel
* .....

The cons of detailing each protocol here is whenever we add another protocol this doc again needs to change. My suggestion is to only have the core flow here and put more details of the protocols on other docs.

Reply:
Makes sense